### DISCUSSION TITLE
FreeSurfer 7.4.1: error with matlab compiler runtime, when trying to do thalamic segmentation (to ultimately use TRACULA)

### DISCUSSION THREAD

**Discussion by emilyharriott** [neurodesk/neurodesk.github.io-hugo#392] [Q&A] [answered]
Dear all, 

I am experimenting with using FreeSurfer (version 7.4.1) on the local neurodesk app, using one locally saved MR image (which I imported into the local neurodesk app using the drag and drop method). 

I successfully ran:

_export SUBJECTS_DIR=/home/jovyan/Desktop
export FS_LICENSE=/home/jovyan/Desktop/license.txt
recon-all -subject testbrain -i /home/jovyan/Desktop/testbraint1.nii.gz -all_

My ultimate goal is to run TRACULA on this image. As the next step in the TRACULA processing pipeline, I then tried to run:

_segmentThalamicNuclei.sh testbrain_

This failed immediately. I got an error that said: _The hippoacmpal/amygdala and brainstem modules require the (free) Matlab runtime. You will need to download the Matlab Compiler Runtime (MCR) for Matlab 2019b. To do so, please run the following command (you might need root permissions): fs_install_mcr R2019b._

In response to that, I ran the suggested line:

_fs_installer_mcr R2019b_

This ran for a while, and I then got an error that said: _Exiting with status 0. End - successful. Finished. mv: inter-device move failed: '/tmp/tmp.8q9mEpXWMe/install-target/v97' to '/opt/freesurfer-7.4.1/MCRv97'; unable to remove target: Read-only file system'_

It seems like I might have a root permissions issue? Hoping to get around any sort of root permissions issue, I ran:

_sudo fs_installer_mcr R2019b_

This failed, as the command was not found.

And now, I'm a bit stuck! I did some googling and looked through these messageboards, but couldn't find anything to help me solve this issue. So, I am hoping someone might be able to help me. I need to run this thalamic segmentation to try to run TRACULA (which is my ultimate goal). How do I get around these (seemingly) root permissions/read only permissions issues, install this mcr thing, or otherwise successfully run segmentThalamicNuclei.sh?

Any assistance or guidance you might be able to provide would be so greatly appreciated. If there is any more information I can provide, please let me know. 

Thank you!
Emily Harriott
Doctoral student, Vanderbilt University (Nashville, TN)

**Comment by stebo85** [comment]
Dear @emilyharriott,

We will add this MCR version and release a new version of the freesurfer container. I let you know when you can test :)

Thank you
Steffen

**Reply by emilyharriott** [reply]
Oh wow, fantastic! Thank you so much! 

Is there any chance you might be able to explain what the problem is? Is it just that this freesurfer version 7.4.1 container that's currently in neurodesk doesn't have this (apparently necessary) MCR version already in it? So this new version of the freesurfer container that you can release, would have this version of MCR in it, and then segmentThalamicNuclei.sh would work? 

And, once it is released and I can test it... do I need to update and/or redownload the local neurodesk app?

Thank you again! I really appreciate your help (and impressively speedy reply!)! :)

**Reply by stebo85** [reply, marked as answer]
Dear @emilyharriott,

Yes, the problem is that this MCR version is not installed inside the freesurfer container. To enable reproducible workflows it is not possible for you to install things in addition. Once this MCR version is installed, segmentThalamicNuclei should work nicely. 

Once the new container is released, you shouldn't need to do anything - I will replace the same container version, because it's only the addition of the matlab runtime environment. Let's see if this works :)

Thank you
Steffen

**Reply by emilyharriott** [reply]
This all sounds wonderful! Thank you so much. I'm looking forward to trying it out!!

**Reply by stebo85** [reply]
And if you don't want to wait for the update of the container: You can use Freesurfer version 7.2.0 - there the matlab runtime environment had the correct version and it broke only in later versions :)

**Reply by stebo85** [reply]
Dear @emilyharriott,

I fixed the MCR version in Freesurfer 7.4.1 and it would be wonderful if you could test it :) To get this fixed version, you need to do one manual step for now until we release a new neurodesk version: In the Application menu is an update button:

<img width="300" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/8a42147c-c987-4f2e-b762-319dd66b81c2">

click that and then start freesurfer from the application menu.

Thank you for finding and reporting this problem :) It is now part of our automated tests.

Thank you
Steffen

**Comment by mydream378** [comment]


**Reply by stebo85** [reply]
yes, that would work - but shouldn't be done in Neurodesk, because the runtime and freesurfer are packacked in containers.

**Reply by giacomofiorin** [reply]
Found this while checking about an error that a colleague has.  This line:
```
sudo chmod -R 777 /usr/local/freesurfer
```
is extremely bad practice. At the very least, you should comment about what that would entail in a shared environment with multiple users.

**Reply by stebo85** [reply]
fully agree - that's a spam comment above proposing this!

**Reply by stebo85** [reply]
It's now deleted
